src/interpreter/vm.rs

```diff
@@ -282,9 +406,11 @@ where
         mut callback: Option<
             impl FnMut(&Cid, &ChainMessage, &ApplyRet) -> Result<(), anyhow::Error>,
         >,
-    ) -> Result<Vec<Receipt>, anyhow::Error> {
+        enable_tracing: bool,
+    ) -> Result<(Vec<Receipt>, Vec<InvocResult>), anyhow::Error> {
         let mut receipts = Vec::new();
         let mut processed = HashSet::<Cid>::default();
+        let mut invoc_results = Vec::new();
```

Comment:
Can we set the expected capacity to reduce re-allocations?

Reply:
I think n is expected to be small here, so not worth the optimisation